src/utils/utils_part.c

```diff
@@ -184,7 +184,7 @@ void ocf_part_set_valid(struct ocf_cache *cache, ocf_part_id_t id,
 			cache->conf_meta->valid_parts_no--;
 			part->config->priority = OCF_IO_CLASS_PRIO_LOWEST;
 			part->config->min_size = 0;
-			part->config->max_size = PARTITION_SIZE_MAX;
+			part->config->max_size = cache->conf_meta->cachelines;
```

Comment:
I think it still isn't correct. If I recall correctly IO classes can be configured with cache device detached. Please add a simple functional test to check how max io class occupancy behaves when configured with cache detached and then what happens after attaching a cache device.